plugins/MauticWebinarBundle/Integration/WebikeoIntegration.php

```diff
@@ -0,0 +1,344 @@
+        return 'JWT';
+    }
+
+    /**
+     * Get the array key for the auth token.
+     *
+     * @return string
+     */
+    public function getAuthTokenKey()
+    {
+        return 'token';
+    }
+
+    /**
+     * Get the keys for the refresh token and expiry.
+     *
+     * @return array
+     */
+    public function getRefreshTokenKeys()
+    {
+        return ['refresh_token'];
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @param bool $inAuthorization
+     */
+    public function getBearerToken($inAuthorization = false)
+    {
+        if (!$inAuthorization && isset($this->keys[$this->getAuthTokenKey()])) {
+            return $this->keys[$this->getAuthTokenKey()];
+        }
+
+        return false;
+    }
+
+    /**
+     * @return string
+     */
+    public function getApiUrl()
+    {
+        return 'https://api.webikeo.com/v1';
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return string
+     */
+    public function getAccessTokenUrl()
+    {
+        return $this->getApiUrl().'/login_check';
+    }
+
+    /**
+     * @return bool
+     */
+    public function authCallback($settings = [], $parameters = [])
+    {
+        if ($this->isAuthorized() && !$settings['use_refresh_token']) {
+            return true;
+        }
+        $autUrl                        = $this->getAccessTokenUrl();
+        $settings['encode_parameters'] = 'json';
+        $parameters['username']        = $this->keys['username'];
+        $parameters['password']        = $this->keys['password'];
+        $error                         = false;
+        try {
+            $response = $this->makeRequest($autUrl, $parameters, 'POST', $settings);
+            if (!isset($response[$this->getAuthTokenKey()])) {
+                $error = $response;
+            } else {
+                $this->extractAuthKeys($response, $this->getAuthTokenKey());
+            }
+        } catch (\Exception $e) {
+            return $e->getMessage();
+        }
+
+        return $error;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAvailableLeadFields($settings = [])
+    {
+        return [
+            'email'              => ['label' => 'Email', 'type' => 'string', 'required' => true],
+            'firstName'          => ['label' => 'First Name', 'type' => 'string', 'required' => true],
+            'lastName'           => ['label' => 'Last Name', 'type' => 'string', 'required' => true],
+            'phone'              => ['label' => 'Phone', 'type' => 'string'],
+            'functionLabel'      => ['label' => 'Position', 'type' => 'string'],
+            'companyLabel'       => ['label' => 'Company', 'type' => 'string'],
+        ];
+    }
+
+    /**
+     * @param array $filters
+     * @param bool  $forSegment
+     *
+     * @return array
+     */
+    public function getWebinars($filters = [], $forSegment = false)
+    {
+        if (empty($filters)) {
+            $filters = ['fromDate' => date('Y-m-d H:i:s')];
```

Comment:
In order to have all the past webinars, you must set a date otherwise it sets the date of today. Which means you won't have the past webinars in your query answer.
`date('Y-m-d H:i:s')` should be replaced by `'1900-01-01 00:00:00'`